codec-dns/src/main/java/io/netty/handler/codec/dns/DnsOpCode.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.dns;
+
+/**
+ * The DNS OpCode as defined in <a href="https://tools.ietf.org/html/rfc2929">RFC2929</a>.
+ */
+public class DnsOpCode implements Comparable<DnsOpCode> {
+
+    /**
+     * The 'Query' DNS OpCode, as defined in <a href="https://tools.ietf.org/html/rfc1035">RFC1035</a>
+     */
+    public static final DnsOpCode QUERY = new DnsOpCode(0x00, "QUERY");
+
+    /**
+     * The 'IQuery' DNS OpCode, as defined in <a href="https://tools.ietf.org/html/rfc1035">RFC1035</a>
+     */
+    public static final DnsOpCode IQUERY = new DnsOpCode(0x01, "IQUERY");
+
+    /**
+     * The 'Status' DNS OpCode, as defined in <a href="https://tools.ietf.org/html/rfc1035">RFC1035</a>
+     */
+    public static final DnsOpCode STATUS = new DnsOpCode(0x02, "STATUS");
+
+    /**
+     * The 'Notify' DNS OpCode, as defined in <a href="https://tools.ietf.org/html/rfc1996">RFC1996</a>
+     */
+    public static final DnsOpCode NOTIFY = new DnsOpCode(0x04, "NOTIFY");
+
+    /**
+     * The 'Update' DNS OpCode, as defined in <a href="https://tools.ietf.org/html/rfc2136">RFC2136</a>
+     */
+    public static final DnsOpCode UPDATE = new DnsOpCode(0x05, "UPDATE");
+
+    public static DnsOpCode valueOf(byte b) {
+        switch (b) {
+        case 0x00:
+            return QUERY;
+        case 0x01:
+            return IQUERY;
+        case 0x02:
+            return STATUS;
+        case 0x04:
+            return NOTIFY;
+        case 0x05:
+            return UPDATE;
+        }
+
+        return new DnsOpCode(b);
+    }
+
+    private final byte byteValue;
+    private final String name;
+    private String text;
+
+    public DnsOpCode(int byteValue) {
```

Comment:
Make the constructor private... People should use `valueOf(...)` for this use-case